### PR TITLE
Adding end_date to scheduled_messages

### PR DESCRIPTION
**Pull request by fentie**
migration to add new end_date field to scheduled messages, and schema update
fixing typo in contributing instructions
adding end date to allowed parameters in scheduled messages controller
updating templates to reflect new field
new spec showing optional creation of scheduled message with end date
new scope to retrieve only active messages (no end date or future end date)
updating message sender to only send active messages, using new scope
updating contributing instructions to reflect code drift
moving to method syntax vs scope. fixing logic bug. adding spec
spec for sending only active messages

### CODE REVIEW

**Comment by gemfarmer** [comment]
Hey @fentie. Thanks for the PR! 

It looks like in addition to adding an end date, you also [modified the env configs](https://github.com/18F/dolores-landingham-bot/pull/159/files#diff-6a3371457528722a734f3c51d9238c13R117). Before merging could you please do the following for deployment and documentation purposes?
- [ ] add `AUTH_DOMAIN: gsa.gov` to the [`manifest.yml`](https://github.com/18F/dolores-landingham-bot/blob/master/manifest.yml#L11) within `dolores-app` and `dolores-staging` 
- [ ] add `AUTH_DOMAIN='gsa.gov'` to the [`.sample.env`](https://github.com/18F/dolores-landingham-bot/blob/master/.sample.env#L4)

**Comment by fentie** [comment]
I didn't actually, all I was doing there was updating CONTRIBUTING.md to reflect changes that were already in master (https://github.com/18F/dolores-landingham-bot/commit/677098e26f6100b530c1df6202d506a671ddde57). `AUTH_DOMAIN` is already set in `.sample.env` too, but it's not yet in `manifest.yml`. I'll add that.

**Comment by gemfarmer** [comment]
@fentie haha you're right, my apologies. Thanks for adding to the documentation! I'll check out the rest of your PR later this afternoon :)

File: spec/models/scheduled_message_spec.rb [hunk 1]
```diff
@@ -23,4 +23,14 @@
       expect(ScheduledMessage.date_time_ordering).to match_array([m1, m2, m3, m4, m5])
     end
   end
+
+  describe 'active messages' do
```
Comment (jessieay) [inline spec/models/scheduled_message_spec.rb]: hey! thanks for adding tests for this. you mind updating this `describe` block to reflect the name of the method? eg:

`describe '.active' do`

File: spec/services/daily_message_sender_spec.rb [hunk 2]
```diff
@@ -16,5 +16,25 @@
 
       expect(message_sender_double).to have_received(:run)
     end
+
+    it "sends only active messages" do
```
Comment (jessieay) [inline spec/services/daily_message_sender_spec.rb]: niiice

File: app/controllers/scheduled_messages_controller.rb [hunk 3]
```diff
@@ -50,6 +50,6 @@ def destroy
   private
 
   def scheduled_message_params
-    params.require(:scheduled_message).permit(:body, :days_after_start, :tag_list, :time_of_day, :title)
+    params.require(:scheduled_message).permit(:body, :days_after_start, :tag_list, :time_of_day, :title, :end_date)
```
Comment (jessieay) [inline app/controllers/scheduled_messages_controller.rb]: mind putting this in alpha order? :)

**Comment by jessieay** [comment]
this is looking awesome! thank you @fentie !

**Comment by fentie** [comment]
You're quite welcome, it was kinda fun. There were a few other things I considered (different display formatting if the message was expired, possibly different behavior for "test send" on an expired message) but I figured that was beyond the scope of this issue.

**Comment by jessieay** [comment]
@fentie interested in your ideas for other improvements, for sure! If you find the time, it would be great if you could open a new GH issue for each one so we can discuss them. :)

**Comment by jessieay** [comment]
Merged to master at f7af4b1ea93fe031eed48dafc72860fe6bc4f7ff  :) :rainbow: 

Thanks again, @fentie !

**Comment by jessieay** [comment]
Looks good on staging, just pushed to production!

**Pull request closed without merging**